app/Http/Controllers/Setup/Controller.php

```diff
@@ -50,37 +57,46 @@ public function database() {
 
     public function test_database() {
         // Attempt to connect to the database with the provided credentials
-        try {
-            $db = new \PDO('mysql:dbname='.request('db_name').';host='.request('db_host'),
-                request('db_username'), request('db_password'));
-            $success = true;
-        } catch(\PDOException $e) {
-            $success = false;
+        if(!empty(request('db_name'))) {
+            try {
+                $db = new \PDO('mysql:dbname='.request('db_name').';host='.request('db_host'),
+                    request('db_username'), request('db_password'));
+                $success = true;
+            } catch(\PDOException $e) {
+                $success = false;
+            }
+        } else {
+            try {
+                DB::RAW('SELECT VERSION();');
+                $success = true;
+            } catch(\Exception $e) {
+                die($e->getMessage());
```

Comment:
This looks like it was left in while debugging.

Reply:
it was 😉

Reply:
I can wrap this up in two methods

TestConnectAdHoc and TestConnect

Would you prefer it fulfilling an interface or just two methods?